### PR TITLE
Update URLs + JSON models for newer derpibooru API

### LINKED ISSUE #12: Migrate to Philomena API
Deadline: March 31st

https://derpibooru.org/forums/meta/topics/old-api-deprecation-notice

### PR DESCRIPTION
**Pull request by nullforce**
WIP PR to discuss changes in derpibooru API

Updated image, tag, search, and reverse search to use new API endpoints and models. I expect there to be more cleanup needed, but overall the calls are generally working.

There doesn't appear to be a new user/profile endpoint. This may have been an oversight when the Philomena conversion was done.

Closes #12

@BytewaveMLP

### CODE REVIEW

File: lib/util/Fetch.ts [hunk 1]
```diff
@@ -455,6 +427,21 @@ export class Fetch {
 		return new Promise<any>((resolve, reject) => {
 			const opts = Object.assign({}, Consts.DEFAULT_REQUEST_OPTS, options);
 
+			if (opts.method === 'POST') {
```
Comment (BytewaveMLP) [inline lib/util/Fetch.ts]: Could be condensed down into `request(opts, ...`; the [`request.METHOD`](https://github.com/request/request#requestmethod) are just aliases to `request({ method: METHOD, ...}, ...` anyway.

File: lib/api/Image.ts [hunk 2]
```diff
@@ -74,19 +74,8 @@ export class Image {
 	 * @type {string}
 	 * @memberof Image
 	 */
-	@JsonProperty('tags', String)
-	public readonly tagString: string = '';
```
Comment (BytewaveMLP) [inline lib/api/Image.ts]: Should probably add a `get tagString` if we want to maintain some backwards comparability. From the other changes, though, and due to some structural changes in the Philomena API, backwards compat may not be so easy.

File: lib/api/TagDetails.ts [hunk 3]
```diff
@@ -13,96 +12,8 @@ import { TagCollection } from '../util/TagCollection';
  */
 @JsonObject
 export class TagDetails {
-	/**
```
Comment (BytewaveMLP) [inline lib/api/TagDetails.ts]: Seems this file isn't referenced by `Tag` anymore; is it worth keeping around?

File: lib/util/Fetch.ts (hunk 1 above)
Comment (nullforce) [reply lib/util/Fetch.ts]: TypeScript complains about `request()` not being a function unless I change the `import * as request 'request';` to `import request = require('request');`

File: lib/api/TagDetails.ts (hunk 3 above)
Comment (nullforce) [reply lib/api/TagDetails.ts]: Removed it.

File: lib/api/Image.ts (hunk 2 above)
Comment (nullforce) [reply lib/api/Image.ts]: Added this and fixed up the documentation comment.

**Pull request merged**